include/hipSYCL/sycl/libkernel/host/group_functions.hpp

```diff
@@ -59,12 +59,12 @@ T *get_local_memory_ptr(Group g, size_t number_elements = 1) {
 }
 
 template<typename T, typename Group>
-void free_local_memory_ptr(Group g, T *local_memory_ptr) {
-  if (sizeof(T) > sizeof(void *)) {
+void free_local_memory_ptr(Group g, T *local_memory_ptr, size_t number_elements = 1) {
+  if (sizeof(T) * number_elements > sizeof(void *)) {
```

Comment:
The way this works still seems like a hack to me ;) But okay, I guess everything will get simpler once we just use the stack for that...

Reply:
Yes using the stack is the next point on my TODO-list.
But the fix is not a hack, it is just the same condition I use when allocating the memory :)

Reply:
My interpretation is that this check is needed because for small required amounts of memory you use the memory of the void pointer itself. That *is* a hack ;)

Reply:
Can't really argue against that statement, but it is faster than allocating memory (almost 40% in the one benchmark run I did).

Reply:
It's not a _hack_ - it's a vaild optimization!

Reply:
It's a hacky optimization that's only necessary because we don't use stack memory for all local allocations yet ;)